### PR TITLE
mismatch config

### PR DESCRIPTION
**Pull request by GiacomoLeoneMaria**

### CODE REVIEW

**Comment by scruffynerf** [comment]
does that config.json code work?  I didn't think that sort of resizing would work.  Happy if I'm wrong, but that really should be tested... the # of tokens being different would mean they wouldn't all match up (unless everything is done with the same setup, which I guess is your goal here)

(the other 2 parts look good to me...BTW)

**Comment by scruffynerf** [comment]
I believe the changes made mentioned in https://github.com/astramind-ai/Auralis/issues/43#issuecomment-2560625168
are another reason the above is a bad idea.

File: src/auralis/models/xttsv2/utils/checkpoint_converter.py [hunk 1]
```diff
@@ -104,12 +106,13 @@ def convert_checkpoint(pytorch_checkpoint_path, output_dir, args):
         download_repo_files("AstraMindAI/xttsv2", os.path.join(output_dir, "core_xttsv2"))
         print(f"Saved XTTSv2 weights to {xtts_path}")
         print(f"XTTSv2 weights: {list(xtts_weights.keys())}")
-
 def main():
     parser = argparse.ArgumentParser(description='Convert PyTorch checkpoint to SafeTensors format')
     parser.add_argument('checkpoint_path', type=str, help='Path to PyTorch checkpoint file')
     parser.add_argument('--output_dir', type=str, default=os.getcwd(),
                         help='Output directory (defaults to current working directory)')
+    parser.add_argument('--tokenizer_size', type=int, default=None,
```
Comment (mlinmg) [inline src/auralis/models/xttsv2/utils/checkpoint_converter.py:114]: Here instead of passing of passing as an argument you should take the vocab size form the embedding layer(check for different name types)

File: src/auralis/models/xttsv2/utils/checkpoint_converter.py [hunk 2]
```diff
@@ -24,14 +24,16 @@ def download_repo_files(repo_id, output_path, exclude_extensions=['*.safetensors
 
 
 
-def convert_checkpoint(pytorch_checkpoint_path, output_dir, args):
```
Comment (mlinmg) [inline src/auralis/models/xttsv2/utils/checkpoint_converter.py:27]: In this file you should create a config.json file, be aware that in the download repo function you should be aware that it is downloading the config.json form out hf repo, and this overrides the config.json file you  should create

File: src/auralis/models/xttsv2/utils/checkpoint_converter.py
Comment (mlinmg) [inline src/auralis/models/xttsv2/utils/checkpoint_converter.py:1]: looks good! In the end we need also a test file to check for edge and custom cases
